From kernel-bounces+gregkh=suse.de@suse.de  Thu Sep 11 02:14:13 2008
From: Hannes Reinecke <hare@suse.de>
Date: Wed, 10 Sep 2008 14:16:09 +0200
Subject: acpi: Fixup typo in acpi_device_remove()
To: kernel@suse.de
Message-ID: <20080910121610.33FA818C734@pentland.suse.de>



The driver data is stored in the acpi device, not the driver
core device.

Signed-off-by: Hannes Reinecke <hare@suse.de>
Cc: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/acpi/scan.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/acpi/scan.c
+++ b/drivers/acpi/scan.c
@@ -384,7 +384,7 @@ static int acpi_device_remove(struct dev
 			acpi_drv->ops.remove(acpi_dev, acpi_dev->removal_type);
 	}
 	acpi_dev->driver = NULL;
-	acpi_driver_data(dev) = NULL;
+	acpi_driver_data(acpi_dev) = NULL;
 
 	put_device(dev);
 	return 0;
